### PR TITLE
isotopologue functions updates

### PR DESCRIPTION
**Pull request by andreavicini**

### CODE REVIEW

File: R/zzz.R [hunk 1]
```diff
@@ -11,8 +11,24 @@
     assign(".ADDUCTS_MULT", add_multi, envir = asNamespace(pkgname))
     assign(".ADDUCTS_ADD", add_add, envir = asNamespace(pkgname))
     
-    substs_hmdb <- utils::read.table(system.file("substitutions/hmdb.txt",
-                                            package = "MetaboCoreUtils"),
-                                sep = "\t", header = TRUE)
-    assign(".SUBSTS_HMDB", substs_hmdb, envir = asNamespace(pkgname))
+    substs_hmdb_neutral <-
```
Comment (jorainer) [inline R/zzz.R:14]: I would suggest to maybe rename the txt files to `hmdb_neutral.txt`, `hmdb_positive.txt` etc and then load all files in the `substitutions` folder (in a loop) and assign each table with `assign(paste0(".", toupper(sub(".txt$", "", basename(fl)))...`

That way we could later add other definitions without the need to change the R code here.

Comment (jorainer) [reply R/zzz.R:14]: Or maybe even better, call the files "substs_hmdb_neutral.txt" - then you just need to add a `"."` before, make upper case and remove the file ending.

File: inst/substitutions/hmdb_subst_negative.txt [hunk 2]
```diff
@@ -0,0 +1,915 @@
+"name"	"md"	"minmass"	"maxmass"	"leftend"	"rightend"	"LBint"	"LBslope"	"UBint"	"UBslope"
```
Comment (jorainer) [inline inst/substitutions/hmdb_subst_negative.txt:1]: maybe better if we remove the columns minmass and maxmass from all files - we don't use them at present, and the smaller the table the faster the subsetting operations.

File: inst/substitutions/hmdb_subst_neutral.txt [hunk 3]
```diff
@@ -0,0 +1,679 @@
+"name"	"md"	"minmass"	"maxmass"	"leftend"	"rightend"	"LBint"	"LBslope"	"UBint"	"UBslope"
```
Comment (jorainer) [inline inst/substitutions/hmdb_subst_neutral.txt:1]: would also remove the minmass and maxmass columns.

File: inst/substitutions/hmdb_subst_positive.txt [hunk 4]
```diff
@@ -0,0 +1,1115 @@
+"name"	"md"	"minmass"	"maxmass"	"leftend"	"rightend"	"LBint"	"LBslope"	"UBint"	"UBslope"
```
Comment (jorainer) [inline inst/substitutions/hmdb_subst_positive.txt:1]: also here, remove minmass maxmass

**Review by jorainer — CHANGES_REQUESTED** [review]
Looks good. I have just some minor change requests. Regarding the "closest" - isn't it that `duplicates = "keep"` will anyway just report (the first) one hit? Maybe not bad if you could try with `duplicates = "keep"`, but then I would create another function (maybe replacing the `isotopes 2` function that, as far as I remember, had no real speed benefits.

File: R/zzz.R (hunk 1 above)
Comment (andreavicini) [reply R/zzz.R:14]: would the code below be correct? Or is there another way to get files in a package besides `list.files`? I get an  error when running `rcmdcheck` ("object '.SUBSTS_HMDB_NEUTRAL' not found") and I wonder if it's wrong to use list.files. Again, beside `rcmdcheck` it works
```suggestion
       txts <- list.files("inst/substitutions", "*.txt", full.names = TRUE)
    for (txt in txts) {
        substs_hmdb <- utils::read.table(txt, sep = "\t", header = TRUE)
        assign(paste0(".", toupper(sub(".txt", "", basename(txt)))),
               substs_hmdb, envir = asNamespace(pkgname))
    }
```

Comment (jorainer) [reply R/zzz.R:14]: I think specifying the path like that `"inst/substitutions"` will not work. You'll need to use:

```r
txts <- dir(system.file("inst", "substitutions", package = "MetaboCoreUtils"), full.names = TRUE, pattern = "txt$")
```

then it *should* (?) hopefully work...

Comment (andreavicini) [reply R/zzz.R:14]: Thanks, I think I've discovered the problem. By removing `"inst" ` from the code you suggested it works. I don't know why but that folder does not exist where the package is installed (although it exists in the repo).

**Pull request merged**